### PR TITLE
feat: add zadd and zrevrange implementations

### PR DESCRIPTION
**Pull request by rishtigupta**
## PR Description:
- Implement `zadd` and `zrevrange` methods
- Add integration tests

## Testing
- Developing the methods by basing them off of the [main branch of client-sdk-php repo](https://github.com/momentohq/client-sdk-php/tree/main) [check composer.json]

## Issue
https://github.com/momentohq/dev-eco-issue-tracker/issues/1033

### CODE REVIEW

File: src/MomentoCacheClient.php [hunk 1]
```diff
@@ -2046,7 +2064,33 @@ public function zRemRangeByScore(string $key, string $start, string $end): Redis
...
-        throw MomentoToPhpRedisExceptionMapper::createCommandNotImplementedException(__FUNCTION__);
+        if ($start >= 0 && $end >= 0 && $start > $end) {
+            return [];
+        }
+        if ($start < 0 && $end < 0 && $start > $end) {
+            return [];
+        }
+
+        // Momento uses exclusive range for end rank, while PhpRedis uses inclusive range
+        $end = $end + 1;
```
Comment (malandis) [inline src/MomentoCacheClient.php]: For the -1 case we need unbounded end, otherwise this will be 0. Let's add a test case for this situation.

Comment (rishtigupta) [reply src/MomentoCacheClient.php]: Doesn't this test case satisfy it:

```
 // Test valid rank range with positive start and negative end
        $elements = self::$client->zRevRange($key, 0, -1);
        $this->assertEquals([$member3, $member2, $member1], $elements, "Elements do not match expected order for mixed ranks");
```

Comment (malandis) [reply src/MomentoCacheClient.php]: Yep that looks good

File: src/MomentoCacheClient.php [hunk 2]
```diff
@@ -2114,19 +2114,24 @@ public function zRemRangeByScore(string $key, string $start, string $end): Redis
...
+        // Handle the unbounded end case (-1 in Redis means unbounded, but we need to handle it in Momento)
+        if ($end == 0) {
+            $end = null;
+        }
+
+        var_dump("start: $start, end: $end");
         $result = $this->client->sortedSetFetchByRank($this->cacheName, $key, $start, $end, SORT_DESC);
         if ($result->asHit()) {
             $elements = $result->asHit()->valuesArray();
+            var_dump($elements);
```
Comment (malandis) [inline src/MomentoCacheClient.php]: Looks like some debug code snuck in

**Review by malandis — APPROVED** [review]

**Pull request merged**